doofinder/includes/settings/class-register-settings.php

```diff
@@ -87,13 +83,7 @@ function () {
 		add_settings_field(
 			$api_key_option_name,
 			__( 'API Key', 'doofinder_for_wp' ),
-			function () use ( $is_indexing, $api_key_option_name ) {
-				if ( $is_indexing ) {
-					$this->render_html_indexing_in_progress();
-
-					return;
-				}
-
+			function () use ( $api_key_option_name ) {

```

Comment:
IMO, this is very ugly and difficult to understand. The anonymous functions running on declaration usually are not very explicit on what they do, at least to me.

Reply:
The wordpress way, in the improvement task I will investigate another way, but you have to pass the function.